modal-go/sandbox.go

```diff
@@ -214,6 +220,44 @@ func (sb *Sandbox) Tunnels(timeout time.Duration) (map[int]*Tunnel, error) {
 	return sb.tunnels, nil
 }
 
+// Poll checks if the Sandbox has finished running.
+// Returns nil if the Sandbox is still running, else returns the exit code.
+func (sb *Sandbox) Poll() (*int32, error) {
+	resp, err := client.SandboxWait(sb.ctx, pb.SandboxWaitRequest_builder{
+		SandboxId: sb.SandboxId,
+		Timeout:   0,
+	}.Build())
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.GetResult() != nil && resp.GetResult().GetStatus() != pb.GenericResult_GENERIC_STATUS_UNSPECIFIED {
+		sb.result = resp.GetResult()
+	}
+
+	return sb.ReturnCode(), nil
+}
```

Comment:
Not sure about this but here's a question on my mind: Is the behavior of this method supposed to be that it still makes a request if `sb.result` / ReturnCode() is saved already from a previous call? I guess I'm questioning this since it doesn't make a ton of sense that the result would change if it's already been retrieved, and maybe it could even expire later on?